### DISCUSSION TITLE
How to enable HTTPS for web console of a cluster on AWS?

### DISCUSSION THREAD

**Discussion by yukccy** [okd-project/okd#1316] [Q&A]
I have followed the steps from [here](https://docs.okd.io/latest/installing/installing_aws/installing-aws-default.html) to deploy a openshift cluster on AWS, and the output looks fine.

As I wanna access the web console with https, I tried to modify the load balancer setting with attaching SSL cert on listeners. However, when I tried to run ''oc whoami" again, it prompted "Unable to connect to the server: x509: certificate is valid for *.mydomain.com, not api.cluster-name.mydomain.com". 

My SSL cert is a wildcard cert, why would it happened? If I shouldn't enable the HTTPS by this way, what's the correct way?

Thanks!

**Comment by vrutkovs** [comment]
>My SSL cert is a wildcard cert, why would it happened?

This is a wildcard cert for `*.mydomain`, which means `cluster-name.mydomain.com` gets valid cert, but not not `api.cluster-name.mydomain.com`.

You need a wildcard cert for `*.cluster-name.mydomain.com` here

**Reply by yukccy** [reply]
I request another SSL cert on ACM for api.cluster-name.mydomain.com and apps.cluster-name.mydomain.com. Now, it prompted "Unable to connect to the server: x509: certificate signed by unknown authority".

The new cert is issued by AWS, why would it happen?

**Comment by yukccy** [comment]
I know there's official ways to apply custom cert. However, as my cert is issued by AWS ACM, I dont have the actual cert file. How can I apply it to the cluster?
